woocommerce/woocommerce-android
# UI to select order statuses filters

Work in progress: #4934 

### Description

The task of filtering orders by order status was too big to create a single PR for it. In fact, this PR is already big enough (sorry about that 😞 ). So I will split it in at least 2 PRs. This is the first part. 
In this PR I've added the essential UI to open filters flow in order list screen and select any combination of order statuses to filter. There are some things that are work in progress (I will indicate this in code) so no need to review them. 

Important pieces missing in this PR that will come in part II here #5087
- Loading order status from APIs call
- Update order list screen with selected filters
- Unit testing for viewModel


### Testing instructions
Open Order List and click the filters button. Play around selecting and unselecting different options, and check the filter screen is updated accordingly.  Behaviors to keep in mind. 

- [x] Order status filters is multi-choice
- [x] When any order status is selected, “All” option should be unselected
- [x] If “All” option is selected, then all the rest of the selected filters should be unselected
- [x] “Clear” button will wipe out all selections and will default to “All” option
- [x] Back button will display a confirmation dialog before discarding selected filters
- [x] If all options are unelected, option "All" option must be checked. This behavior is not included in this PR, but is fixed here #5087

https://user-images.githubusercontent.com/2663464/137933974-f6218c3d-011e-474c-a8af-237327632269.mp4
